x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -195,7 +193,7 @@ export const transformOrBulkError = (
   ruleStatus?: unknown
 ): Partial<OutputRuleAlertRest> | BulkError => {
   if (isAlertType(alert)) {
-    if (isRuleStatusFindType(ruleStatus)) {
+    if (isRuleStatusFindType(ruleStatus) && ruleStatus?.saved_objects.length > 0) {
```

Comment:
Looking at the `isRuleStatusFindType(ruleStatus)` you shouldn't have to do a double tap with `ruleStatus?` as that is check should return false and never an undefined or null if `ruleStatus` is null?

But it's ok to keep as the line below has it and you're doing a back port. But with all the tests you could push in a null to this test or the `isRuleStatusFindType` and see that it will return a boolean `false` and not a true for `ruleStatus`

Reply:
Ah yes that is true. I also can probably remove the nullish coalescing and just grab the first object from the saved_objects array. With the `isRuleStatusFindType` I am pretty sure that should remove any need for having the nullish coalescing.

Reply:
Working through this fix I found a pre-existing bug which I will open a fix for in a separate PR.